pkg/plugins/manager/loader/angulardetector/angulardetector.go

```diff
@@ -1,46 +1,60 @@
 )
 
-var (
-	_ detector = &containsBytesDetector{}
-	_ detector = &regexDetector{}
-)
-
-// detector implements a check to see if a plugin uses Angular.
-type detector interface {
-	// Detect takes the content of a moduleJs file and returns true if the plugin is using Angular.
-	Detect(moduleJs []byte) bool
-}
-
-// containsBytesDetector is a detector that returns true if module.js contains the "pattern" string.
-type containsBytesDetector struct {
-	pattern []byte
+// defaultDetectors contains all the detectors to detect Angular plugins.
+// They are executed in the specified order.
+var defaultDetectors = []detector{
+	&containsBytesDetector{pattern: []byte("PanelCtrl")},
+	&containsBytesDetector{pattern: []byte("QueryCtrl")},
+	&containsBytesDetector{pattern: []byte("app/plugins/sdk")},
+	&containsBytesDetector{pattern: []byte("angular.isNumber(")},
+	&containsBytesDetector{pattern: []byte("editor.html")},
+	&containsBytesDetector{pattern: []byte("ctrl.annotation")},
+	&containsBytesDetector{pattern: []byte("getLegacyAngularInjector")},
+
+	&regexDetector{regex: regexp.MustCompile(`['"](app/core/utils/promiseToDigest)|(app/plugins/.*?)|(app/core/core_module)['"]`)},
+	&regexDetector{regex: regexp.MustCompile(`from\s+['"]grafana\/app\/`)},
+	&regexDetector{regex: regexp.MustCompile(`System\.register\(`)},
 }
 
-// Detect returns true if moduleJs contains the byte slice d.pattern.
-func (d *containsBytesDetector) Detect(moduleJs []byte) bool {
-	return bytes.Contains(moduleJs, d.pattern)
+// newDefaultStaticDetectorsProvider returns a new staticDetectorsProvider with the default (hardcoded) angular
+// detection patterns (defaultDetectors)
+func newDefaultStaticDetectorsProvider() detectorsProvider {
+	return &staticDetectorsProvider{detectors: defaultDetectors}
 }
 
-// regexDetector is a detector that returns true if the module.js content matches a regular expression.
-type regexDetector struct {
-	regex *regexp.Regexp
+// newRemoteInspector returns the default remote Inspector, which is a PatternsListInspector that will:
+//  1. Try to get the Angular detectors from GCOM
+//  2. If it fails, it will use the hardcoded detections provided by defaultDetectors.
+func newRemoteInspector(cfg *config.Cfg) (Inspector, error) {
+	remoteProvider, err := newGCOMDetectorsProvider(cfg.GrafanaComURL, defaultGCOMDetectorsProviderTTL)
+	if err != nil {
+		return nil, fmt.Errorf("newGCOMDetectorsProvider: %w", err)
+	}
+	return &PatternsListInspector{
+		detectorsProvider: sequenceDetectorsProvider{
+			remoteProvider,
+			newDefaultStaticDetectorsProvider(),
+		},
+	}, nil
 }
 
-// Detect returns true if moduleJs matches the regular expression d.regex.
-func (d *regexDetector) Detect(moduleJs []byte) bool {
-	return d.regex.Match(moduleJs)
+// newHardcodedInspector returns the default Inspector, which is a PatternsListInspector that only uses the
+// hardcoded (static) angular detection patterns.
+func newHardcodedInspector() (Inspector, error) {
```

Comment:
since we are using the same approach than when retrieving the public manifest key, it would be nice to follow the same approach. In that case:

 - Define the interface that both the hardcoded and the remote inspectors will follow in the `plugins` package (`ifaces.go`). BTW, there we use  `static` for what you can `hardcoded` and `dynamic` for what you call `remote`. Probably it's better to use the same naming.
 - Here within the plugin manager, you expose the default behavior (static/hardcoded). You can see `static_retriever.go` for how it's implemented there.
 - Then, in `pluginsintegrations` you define the dynamic/remote version of the service and the wrapper that instantiates the static or the dynamic version based on configuration. The existing package there that does this is called `keyretriever`.

Reply:
:+1:  sounds good, I will refactor it so it's similar to signing keys retrieval

Reply:
This will go into a follow up PR, so I'm marking this one as resolved